File: typescript/leap/leap.ts
const year = {
  value: 0,
  isDivisibleBy: function (divisor: number): boolean {
    return this.value % divisor === 0;
  },
};

function isLeapYear(n: number): boolean {
  year.value = n;

  return (
    year.isDivisibleBy(4) &&
    (!year.isDivisibleBy(100) || year.isDivisibleBy(400))
  );
}

export default isLeapYear;

File: julia/leap/leap.jl
"""
    is_leap_year(year)

Return `true` if `year` is a leap year in the gregorian calendar.

"""
is_diviseble_by = fn(x, y) = x % y == 0 

function is_leap_year(year)
  return is_diviseble_by(year, 4) && (!is_diviseble_by(year, 100) || is_diviseble_by(year, 400))
end

